lib/experimental/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -182,9 +182,9 @@ public static function get_merged_data( $origin = 'custom' ) {
 		}
 
 		$result = new WP_Theme_JSON_Gutenberg();
-		$result->merge( static::get_core_data() );
-		$result->merge( static::get_block_data() );
-		$result->merge( static::get_theme_data() );
+		$result->merge( apply_filters( 'global_styles_core', static::get_core_data() ) );
+		$result->merge( apply_filters( 'global_styles_blocks', static::get_block_data() ) );
+		$result->merge( apply_filters( 'global_styles_theme', static::get_theme_data() ) );
 		if ( 'custom' === $origin ) {
 			$result->merge( static::get_user_data() );
 		}
```

Comment:
Just in the line below there's `set_spacing_sizes`, a configuration for spacing presets that does not follow the normal algorith (core > theme > user), so it wouldn't be filterable. I need to figure out how that works and whether it can work like any other `theme.json` setting. See https://github.com/WordPress/gutenberg/pull/41527/files#r966773252